giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/tokens/impl/MitreidAccessTokenChecker.java

```diff
@@ -0,0 +1,58 @@
+package edu.asu.diging.gilesecosystem.web.aspects.access.tokens.impl;
+
+import io.jsonwebtoken.MalformedJwtException;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.gilesecosystem.web.aspects.access.openid.google.CheckerResult;
+import edu.asu.diging.gilesecosystem.web.aspects.access.openid.google.ValidationResult;
+import edu.asu.diging.gilesecosystem.web.aspects.access.tokens.IChecker;
+import edu.asu.diging.gilesecosystem.web.exceptions.AppMisconfigurationException;
+import edu.asu.diging.gilesecosystem.web.exceptions.IdentityProviderMisconfigurationException;
+import edu.asu.diging.gilesecosystem.web.exceptions.InvalidTokenException;
+import edu.asu.diging.gilesecosystem.web.exceptions.ServerMisconfigurationException;
+import edu.asu.diging.gilesecosystem.web.tokens.IApiTokenContents;
+import edu.asu.diging.gilesecosystem.web.tokens.INimbusTokenService;
+import edu.asu.diging.gilesecosystem.web.tokens.impl.IntrospectTokenService;
+
+@Service
+public class MitreidAccessTokenChecker implements IChecker {
```

Comment:
updated javadoc
I'll write test cases for IntrospectTokenService and MitreidAccessTokenChecker as separate subtast under this task.